code/game/atom_vv.dm

```diff
@@ -57,33 +57,42 @@
 			return
 
 		if(!reagents)
-			var/amount = input(usr, "Specify the reagent size of [src]", "Set Reagent Size", 50) as num
+			var/amount = tgui_input_number(usr, "Specify the reagent size of [src]", "Set Reagent Size", 50, 1000000, 1)
+			if(!amount)
+				return
+			create_reagents(amount)
+
+		var/chosen_id
+		var/list/reagent_options = sortAssoc(GLOB.chemical_reagents_list)
+		var/response = tgui_input_list(usr, "Choose a method.", "Add reagents", list("Enter ID", "Choose ID"))
+		switch(response)
+			if("Enter ID")
+				var/valid_id = FALSE
+				chosen_id = tgui_input_text(usr, "Enter the ID of the reagent you want to add.", "Choose a reagent")
```

Comment:
Is there any reason to not just make this a list of all the reagents since its searchable with TGUI now?